src/bp/ui-shared/src/Contents/Components/Fields/Select.tsx

```diff
@@ -1,22 +1,65 @@
-import React, { FC } from 'react'
+import { FormData, FormDynamicOptions, FormField, FormOption } from 'botpress/sdk'
+import _ from 'lodash'
+import React, { FC, Fragment, useEffect, useState } from 'react'
 
+import { lang } from '../../../translations'
 import Dropdown from '../../../Dropdown'
 import style from '../style.scss'
 import { FieldProps } from '../typings'
 
 interface SelectProps extends FieldProps {
-  options: { value: string; label: string }[]
+  data: FormData
+  field: FormField
+  bp?: any
+  parent?
+  printField?
 }
 
-const Select: FC<SelectProps> = ({ onChange, options, value }) => (
-  <Dropdown
-    filterable={false}
-    className={style.formSelect}
-    items={options}
-    defaultItem={value}
-    rightIcon="chevron-down"
-    onChange={option => onChange?.(option.value)}
-  />
-)
+const Select: FC<SelectProps> = ({ onChange, printField, parent, field, data, bp }) => {
+  const [options, setOptions] = useState<FormOption[]>([])
+
+  useEffect(() => {
+    if (field.dynamicOptions) {
+      // tslint:disable-next-line: no-floating-promises
+      loadListElements()
+    } else {
+      setOptions(field.options as FormOption[])
+    }
+  }, [])
+
+  const loadListElements = async () => {
+    const { endpoint, path, valueField, labelField } = field.dynamicOptions as FormDynamicOptions
+
+    try {
+      const { data } = await bp.axios.get(
```

Comment:
On the studio, axios is configured globally so you don't need to get bp.axios. Just import axios normally

Reply:
In the studio yes, but not in the ui-shared, since the components may be imported from the studio or modules, axios might not be configured globally (qna had that issue and is using the same components)